### PR TITLE
Update MUSCLE to v5.1

### PR DESCRIPTION
**Pull request by nsoranzo**
Latest building instructions: https://github.com/rcedgar/muscle/wiki/Building-MUSCLE

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by nsoranzo** [comment]
Marking as WIP since the author [mentioned](https://github.com/rcedgar/muscle/issues/6#issuecomment-879249350) that "It's premature to update muscle in widely-used distributions such as Debian and Bioconda, this should be regarded as an experimental/beta release which has yet been widely tested in practice."

File: recipes/muscle/meta.yaml [hunk 1]
```diff
@@ -1,33 +1,35 @@
 {% set name = "muscle" %}
-{% set version = "3.8.1551" %}
+{% set version = "5.0.1278" %}
```
Comment (colinbrislawn) [inline recipes/muscle/meta.yaml:2]: v5.0.1428 is out as a tagged release!
https://github.com/rcedgar/muscle/releases/tag/v5.0.1428

File: recipes/muscle/meta.yaml [hunk 2]
```diff
@@ -1,33 +1,35 @@
...
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  md5: 1b7c9661f275a82d3cf708f923736bf8
-  url: http://www.drive5.com/muscle/muscle_src_3.8.1551.tar.gz
-  patches:
-    - osx-makefile.patch
+  git_url: https://github.com/rcedgar/muscle
+  git_rev: 28089b1ece7c963a1f377db314d7c70ad0c08d33
```
Comment (colinbrislawn) [inline recipes/muscle/meta.yaml:10]: git_rev: 43fde3c6e10cb0aa065a2c87075dd9e3672b4a79
for v5.0.1428

**Review by colinbrislawn — COMMENTED** [review]
bump version and `git_rev` for version `v5.0.1428`

I have not yet tested this new build

Thanks for submitting this recipe!

File: recipes/muscle/meta.yaml (hunk 1 above)
Comment (nsoranzo) [reply recipes/muscle/meta.yaml]: Thanks @colinbrislawn, I've updated the PR accordingly.

**Comment by nsoranzo** [comment]
Blocking issue: https://github.com/rcedgar/muscle/issues/8

**Comment by colinbrislawn** [comment]
OSX build is failing but I'm not getting much from this error

Any ideas? I have an Intel Mac I could try building this on if that's helpful...
```
17:17:02 BIOCONDA INFO (OUT) mkdir -p o/
17:17:02 BIOCONDA INFO (OUT) x86_64-apple-darwin13.4.0-clang++ -fopenmp -ffast-math -msse -mfpmath=sse -O3 -DNDEBUG -c -o o/addconfseq.o addconfseq.cpp
17:17:07 BIOCONDA ERROR COMMAND FAILED (exited with 1): /Users/distiller/project/miniconda/bin/conda build --override-channels --no-anaconda-upload -c conda-forge -c bioconda -c defaults -e /Users/distiller/project/miniconda/conda_build_config.yaml -e /Users/distiller/project/miniconda/lib/python3.7/site-packages/bioconda_utils/bioconda_utils-conda_build_config.yaml recipes/muscle/meta.yaml

17:17:07 BIOCONDA ERROR BUILD FAILED recipes/muscle
```

**Comment by nsoranzo** [comment]
@colinbrislawn The real error is:
```
17:17:00 BIOCONDA INFO (ERR) In file included from addconfseq.cpp:1:
17:17:00 BIOCONDA INFO (ERR) In file included from ./muscle.h:21:
17:17:00 BIOCONDA INFO (ERR) ./myutils.h:31:10: fatal error: 'omp.h' file not found
17:17:00 BIOCONDA INFO (ERR) #include <omp.h>
17:17:00 BIOCONDA INFO (ERR)          ^~~~~~~
17:17:00 BIOCONDA INFO (ERR) 1 error generated.
17:17:00 BIOCONDA INFO (ERR) make: *** [Makefile:183: o/addconfseq.o] Error 1
```

It may be that muscle's Makefile is overriding some CFLAGS/LDFLAGS removing the include path of `omp.h` . I don't have time to investigate for a couple of days, so feel free to have a try debugging it!

**Comment by rdenise** [comment]
New release of the version 5 released on github: v5.1
On the release note they said: Improved build system, implement github actions/CI, fix crashing bug when gaps in input sequences.

It will probably help for the build failed error that this recipe is stuck on.

(If I said something stupid forgot about my comment)

**Comment by colinbrislawn** [comment]
Cool!

Want to update this PR to new release, or submit a new one?

**Comment by rdenise** [comment]
I don't know how to do it but if someone explains, I can try to do it

**Comment by nsoranzo** [comment]
I'll give another try updating this PR.

**Comment by colinbrislawn** [comment]
Screen Shot 2022-02-09 at 10 34 26 AM

I'm betting nsoranzo is best situated to push to that branch, so let's wait for them to update.
(You can submit PRs to other people's PRs, but let's keep this simple, 🙃 )

**Comment by nsoranzo** [comment]
@BiocondaBot please add label

**Review by dpryan79 — APPROVED** [review]

**Comment by dpryan79** [comment]
Wow, well done everyone!

**Pull request merged**